app-server/app/src/main/scala/user/User.scala

```diff
@@ -2,40 +2,65 @@ package com.azavea.rf.user
 
 import scala.concurrent.{Future, ExecutionContext}
 import scala.util.{Try, Success, Failure}
+import scala.language.postfixOps
 import java.sql.Timestamp
-import slick.lifted.{ColumnOrdered, TableQuery, Query}
+import slick.lifted.Query
 import com.lonelyplanet.akka.http.extensions.{PageRequest, Order}
-import de.choffmeister.auth.common.JsonWebToken
 
-import com.azavea.rf.datamodel.latest.schema.tables.{Users, Organizations}
-import com.azavea.rf.datamodel.latest.schema.tables.UsersRow
+import com.azavea.rf.datamodel.latest.schema.tables.{Users, Organizations, UsersToOrganizations}
+import com.azavea.rf.datamodel.latest.schema.tables.{UsersRow, UsersToOrganizationsRow}
 import com.azavea.rf.utils.{Database,PaginatedResponse}
 
+object Roles extends Enumeration {
+  val Uploader = Role("uploader")
+  val Viewer = Role("viewer")
+  val Admin = Role("admin")
+
+  def Role(name: String): Value with Matching =
+    new Val(nextId, name) with Matching
+
+  def unapply(s: String): Option[Value] =
+    values.find(s == _.toString)
+
+  trait Matching {
+    def unapply(s: String): Boolean =
+      (s == toString)
+  }
+}
 
 /**
-  * Instantiated from a POST request to create a user
+  * Instantiated from a POST request to create a user. Typically used with the default public org.
   */
-case class UsersRowCreate(authId: String, organizationId: java.util.UUID) {
-  def toUsersRow(): UsersRow = {
+case class UsersRowCreate(id: String, organizationId: java.util.UUID, role: Option[String] = Some("uploader")) {
+  def toUsersOrgTuple(): (UsersRow, UsersToOrganizationsRow)= {
     val now = new Timestamp((new java.util.Date()).getTime())
     val newUUID = java.util.UUID.randomUUID
-    UsersRow(
-      id=newUUID,
-      authId=authId,
-      organizationId=organizationId
+    val user = UsersRow(
+      id=id
     )
+    val defaultedRole = role getOrElse { "null" } match {
```

Comment:
Two things here:

1. The `getOrElse` and immediately throwing seems like not a great pattern. Maybe something like this instead:
    ```scala
        val defaultedRole = role match {
          case Some(Roles(role)) => role
          case Some(s: String) => throw new Exception( "\"" + s + "\" is not a valid user Role")
          case None => Roles.Uploader
        }

2. This should throw a more specific exception if a bad role is passed through and then an exception handler should be added to make it a more general client error (`400`) instead of an internal server error.